versions/3.0.4.md

```diff
@@ -383,7 +380,7 @@ An object representing a Server.
 
 | Field Name | Type | Description |
 | ---- | :----: | ---- |
-| <a name="server-url"></a>url | `string` | **REQUIRED**. A URL to the target host. This URL supports Server Variables and MAY be relative, to indicate that the host location is relative to the location where the OpenAPI document is being served. Variable substitutions will be made when a variable is named in `{`braces`}`. |
+| <a name="server-url"></a>url | `string` | **REQUIRED**. A URL to the target host. This URL supports Server Variables and MAY be relative, to indicate that the host location is relative to the location where the entry document of the OpenAPI Description is being served. Variable substitutions will be made when a variable is named in `{`braces`}`. |
```

Comment:
I think this might also be document.  It doesn't matter for Server Objects under the OpenAPI Object's `servers` field, and if there are no such Server Objects, then we're basically "pretending" that there is one with a `url` based on the entry document's location.

But this gets weird when you have a Path Item Object in a referenced document that defines its own `servers`, or that has Operation Objects that define their own `servers` (or a Link Object that defines `server`).  I think those ought to be relevant to the location of the document in which they appear.

In "Resolving Implicit Connections", I hand-waved the Servers problem on the grounds that only the entry document's Paths Object is relevant.  But with a shared Path Item Object under the Components Object of a components-only "syntactically complete" document, do you look up the default by current document or by entry document?  That might actually need clarification.

Reply:
>In "Resolving Implicit Connections", I hand-waved the Servers problem on the grounds that only the entry document's Paths Object is relevant. But with a shared Path Item Object under the Components Object of a components-only "syntactically complete" document, do you look up the default by current document or by entry document? That might actually need clarification.

It would seem logical to me then the lookup happens on the entry document.

Reply:
This ought to follow RFC3986 which would go:

* RFC3986 §5.1.2 enclosing context (OAS 3.0 has always violated RFC3986 by skipping this)
* RFC3986 §5.1.3 retrieval URL of the resource, which in this case would be the _current document_
* RFC3986 §5.1.4 application-specific default, which could be anything we want

Do we state anywhere how to locate Server Objects from referenced (non-entry) documents that are OpenAPI Documents and therefore might have an OpenAPI Object (and `servers` field) in the _document_ root that is _not_ the _OAD_ root (which is always the entry document)?

Reply:
Agreed to keep this as-is, but it needs more thought for 3.1

Reply:
@darrelmiller and I had a discussion and agreed that the original wording ("OpenAPI document" with a small "d") is most logically read as "current document" and should be retained.

@darrelmiller can you confirm this as we're reversing what was agreed in the call?